### PR TITLE
Bug/dynamic activation url

### PR DESCRIPTION
**Pull request by tomivm**
Reading origin header.

### CODE REVIEW

File: api/mail/email-verification.js [hunk 1]
```diff
@@ -379,21 +378,22 @@ module.exports = function(mongoose) {
...
    * @param {string} email - the user's email address.
+   * @param {string} domain - the user's domain.
    * @param {string} url - the unique url generated for the user.
    * @param {function} callback - the callback to pass to Nodemailer's transporter
    */
-  var sendVerificationEmail = function(email, url, callback) {
+  var sendVerificationEmail = function(email, domain, url, callback) {
     var r = /\$\{URL\}/g;
-
+    var d = /DOMAIN/;
```
Comment (martinbedouret) [inline api/mail/email-verification.js:387]: use the same pattern as URL, for consistency 
`/\$\{DOMAIN\}/g`

File: api/mail/index.js [hunk 2]
```diff
@@ -39,10 +39,9 @@ module.exports = function(locale) {
         (text =
           'Thanks for signup to Cboard!. Please confirm your account by clicking the following link: ${URL}');
   }
-
   nev.configure(
     {
-      verificationURL: 'https://app.cboard.io/activate/${URL}',
+      verificationURL: 'DOMAIN/activate/${URL}',
```
Comment (martinbedouret) [inline api/mail/index.js:44]: use the same pattern as URL, for consistency 
`${DOMAIN}`

File: api/mail/index.js [hunk 3]
```diff
@@ -39,10 +39,9 @@ module.exports = function(locale) {
...
       resetPasswordURL: 'https://app.cboard.io/reset/${USERID}/${URL}',
```
Comment (martinbedouret) [inline api/mail/index.js:45]: we should also implement for this link

File: api/mail/email-verification.js [hunk 4]
```diff
@@ -473,10 +473,9 @@ module.exports = function(mongoose) {
    * @func resendVerificationEmail
    * @param {object} email - the user's email address
```
Comment (martinbedouret) [inline api/mail/email-verification.js:474]: add new parameter in decorators

**Review by martinbedouret — CHANGES_REQUESTED** [review]

File: api/mail/email-verification.js (hunk 4 above)
Comment (tomivm) [reply api/mail/email-verification.js:474]: thanks for your feedback!

**Review by martinbedouret — APPROVED** [review]

File: api/mail/email-verification.js [hunk 5]
```diff
@@ -379,21 +378,21 @@ module.exports = function(mongoose) {
...
    */
-  var sendVerificationEmail = function(email, url, callback) {
+  var sendVerificationEmail = function(email, domain, url, callback) {
     var r = /\$\{URL\}/g;
-
+    var d = /\$\{DOMAIN\}/g;
     // inject newly-created URL into the email's body and FIRE
     // stringify --> parse is used to deep copy
-    var URL = options.verificationURL.replace(r, url),
+    var URL = options.verificationURL.replace(d, domain).replace(r, url);
```
Comment (martinbedouret) [inline api/mail/email-verification.js:390]: something is wrong. you replaced the comma by a semicolon!

**Review by martinbedouret — CHANGES_REQUESTED** [review]

**Review by martinbedouret — APPROVED** [review]

**Pull request merged**